File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=0,ans=nums[0];
        for(int i=0; i<nums.length; i++){
            sum+=nums[i];
            if(sum>ans)ans=sum;
            if(sum<0)sum=0;
        }
        return ans;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN;
        int currentSum = 0;
        int nmax=INT_MIN;
        for (int i = 0; i<nums.size(); i++){
            nmax = max(nums[i],nmax);
            currentSum += nums[i];
            if(currentSum<0){
                currentSum = 0;
            }
            maxSum = max(currentSum,maxSum);
        }
        if(maxSum==0){
            return nmax;
        }
        return maxSum;
    }
};